### PR TITLE
[URL] Ensure that path-only URLs do not have their paths erased by th…

### PR DESCRIPTION
**Pull request by karwa**
…e pathname setter.

See: https://github.com/whatwg/url/pull/582

### CODE REVIEW

**Comment by karwa** [comment]
The first added test (special URLs) is just documenting the status-quo, since there didn't seem to be a test for what happens if you set an empty string.

The second one shows the new behaviour I propose on adding to the standard.

**Review by domenic — APPROVED** [review]
Confirmed that the third test added here fails without the change and passes with it, in jsdom/whatwg-url.

File: url/resources/setters_tests.json [hunk 1]
```diff
@@ -1571,6 +1571,33 @@
...
+        },
+        {
+            "comment": "Path-only URLs cannot have their paths erased",
+            "href": "foo:/some/path",
+            "new_value": "",
+            "expected": {
+                "href": "foo:/",
+                "pathname": "/"
+            }
+        },
```
Comment (TimothyGu) [inline url/resources/setters_tests.json:1600]: Can we also add a test where the new `pathname` value is non-empty but rootless, like `"test"`? A root should be automatically created in that case.

**Comment by annevk** [comment]
Ah, I merged the specification PR but I missed the outstanding comment here. @karwa do you still want to add that additional test?

File: url/resources/setters_tests.json (hunk 1 above)
Comment (achristensen07) [reply url/resources/setters_tests.json:1600]: Could we also add a test setting a path on a URL without a host?  My initial late-night implementation of this change would've failed that test.  Like this:

+        {
+            "comment": "Non-special URLs can have their paths erased",
+            "href": "foo:///some/path",
+            "new_value": "",
+            "expected": {
+                "href": "foo://",
+                "pathname": ""
+            }
+        },

**Comment by annevk** [comment]
I added the additional requested tests, once someone reviewed them this should be good to go.

**Review by TimothyGu — APPROVED** [review]

**Comment by karwa** [comment]
Apologies, I missed this - thanks for adding the tests, @annevk

**Pull request merged**